server/resourcemanager/testutil/operations_list.go

```diff
@@ -1,109 +1,251 @@
 package testutil
 
 import (
+	"encoding/json"
 	"fmt"
 	"net/http"
 	"net/http/httptest"
 	"strings"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-func buildListRequest(rt string, ct contentType, testServer *httptest.Server, t *testing.T) *http.Request {
-	url := fmt.Sprintf("%s/%s/", testServer.URL, strings.ToLower(rt))
+func buildQueryString(params map[string]string) string {
+	queryString := ""
 
-	req, err := http.NewRequest(http.MethodGet, url, nil)
-	require.NoError(t, err)
-	return req
-}
+	if len(params) == 0 {
+		return queryString
+	}
 
-const OperationListNoResults Operation = "ListNoResults"
+	formattedParams := []string{}
 
-type listNoResultsOperation struct{}
+	for key, value := range params {
+		formattedParams = append(formattedParams, fmt.Sprintf("%s=%s", key, value))
+	}
 
-func (op listNoResultsOperation) postAssert(t *testing.T, ct contentType, rt ResourceTypeTest, testServer *httptest.Server) {
+	return "?" + strings.Join(formattedParams, "&")
 }
 
-func (op listNoResultsOperation) buildRequest(t *testing.T, testServer *httptest.Server, ct contentType, rt ResourceTypeTest) *http.Request {
-	return buildListRequest(
-		rt.ResourceType,
-		ct,
-		testServer,
-		t,
-	)
-}
+func getSafeSortFields(sortFields []string) []string {
```

Comment:
a nil slice in is safe to use, no need for this func: https://go.dev/play/p/ULVYg48l1R-

Reply:
After our discussion yesterday, it not makes sense for this function anymore, I deleted it.